src/lib/file_upload_helper.py

```diff
@@ -0,0 +1,159 @@
+
+    @staticmethod
+    def is_initialized():
+        if instance is None:
+            return False
+
+        return True
+
+    def __init__(self, upload_temp_dir, upload_dir, uuid_api_url):
+        global instance
+        self.upload_temp_dir = file_helper.ensureTrailingSlash(upload_temp_dir)
+        self.upload_dir = file_helper.ensureTrailingSlash(upload_dir)
+        self.uuid_api_url = uuid_api_url
+        if instance is None:
+            instance = self
+
+    def save_temp_file(self, file):
+        temp_id = self.get_temp_file_id()
+        file_dir = self.__get_temp_file_dir(temp_id)
+
+        # Use pathlib to create dir instead of file_helper.mkDir
+        pathlib.Path(file_dir).mkdir(parents=True, exist_ok=True)
+
+        file.save(os.path.join(file_dir, secure_filename(file.filename)))
+
+        return temp_id
+
+    def __get_temp_file_dir(self, temp_id):
+        return self.upload_temp_dir + temp_id + os.sep
+
+    def get_temp_file_id(self, iteration=0):
+        if iteration == 100:
+            raise Exception("Unable to get a temporary file id after 100 attempts")
+        rid = ''
+        for _ in range(20):
+            rid = rid + secrets.choice(ID_CHARS)
+        while os.path.exists(self.__get_temp_file_dir(rid)):
+            rid = self.get_temp_file_id(iteration=iteration + 1)
+
+        return rid
+
+    def commit_file(self, temp_file_id, entity_uuid, user_token):
+        logger.debug(temp_file_id)
+
+        file_temp_dir = self.__get_temp_file_dir(temp_file_id.strip())
+        if not os.path.exists(file_temp_dir):
+            raise Exception("Temporary file with id " + temp_file_id + " does not have a temp directory.")
+
+        fcount = 0
+        temp_file_name = None
+        for tfile in listdir(file_temp_dir):
+            fcount = fcount + 1
+            temp_file_name = tfile
+
+        if fcount == 0:
+            raise Exception("File not found for temporary file with id " + temp_file_id)
+        if fcount > 1:
+            raise Exception("Multiple files found in temporary file path for temp file id " + temp_file_id)
+
+        file_from_path = file_temp_dir + temp_file_name
+        file_to_dir = self.upload_dir + entity_uuid + os.sep
+
+        # get a uuid for the file
+        checksum = hashlib.md5(open(file_from_path, 'rb').read()).hexdigest()
+        filesize = os.path.getsize(file_from_path)
+        headers = {'Authorization': 'Bearer ' + user_token, 'Content-Type': 'application/json'}
+        data = {}
+        data['entity_type'] = 'FILE'
+        data['parent_ids'] = [entity_uuid]
+        file_info = {}
+        file_info['path'] = file_to_dir + '<uuid>' + os.sep + temp_file_name
+        file_info['checksum'] = checksum
+        file_info['base_dir'] = 'INGEST_PORTAL_UPLOAD'
+        file_info['size'] = filesize
+        data['file_info'] = [file_info]
+        response = requests.post(self.uuid_api_url + '/uuid', json=data, headers=headers, verify=False)
```

Comment:
@ChuckKollar @maxsibilla my main concern is this helper file. 
- HuBMAP already has one: https://github.com/hubmapconsortium/ingest-api/blob/port_metadata_validate_endpoint/src/file_upload_helper.py and it's being used by several other places.
- The one that SenNet uses currently won't compatible with HuBMAP, mainly due to this `/uuid`, which is not in the HuBMAP version since we have that in the configuration.

I propose:
- We keep using the existing `file_upload_helper.py` that has been used by other modules in HuBMAP ingest-api.
- Move the `src/lib`  package to under `src/routes/validation` since they are specific to this metadata validation.

Reply:
Joe, I think that your changes have been implemented and tested. I have also expanded the README file which gives examples of ways of testing this endpoint using curl
Thanks Charles